Assets/Scripts/States/PatrollingState.cs

```diff
@@ -6,30 +6,39 @@ public class PatrollingState : TankState
     Vector3 moveto;
     float timeelasped = 3;
     bool enterState = false;
-    Transform target = GameObject.FindGameObjectWithTag("Player").transform;
+    Transform target;
     private void Update()
     {
         //print(enterState);
         if(enterState)
         {
             MovePatroll();
         }
-        if(Vector3.Distance(enemyBehaviour.transform.position,target.position) < 2.5f)
+        if(Vector3.Distance(enemyBehaviour.transform.position,target.position) < 5)

```

Comment:
you should not be doing distance checks - instead use colliders of different sizes around the enemy to detect the player